src/core/shape_inference/include/multinomial_shape_inference.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "dimension_util.hpp"
+#include "openvino/op/multinomial.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v13 {
+template <class TShape, class TRShape = result_shape_t<TShape>>
+std::vector<TRShape> shape_infer(const Multinomial* op,
```

Comment:
Add unit test to simple call using intel_cpu::StaticShape to check if it will compile for this type of shapes.

Reply:
Added!